channel.go

```diff
@@ -621,6 +628,9 @@ func (ch *Channel) getMinConnectionState() connectionState {
 func (ch *Channel) connectionCloseStateChange(c *Connection) {
 	ch.removeClosedConn(c)
 	if peer, ok := ch.rootPeers().Get(c.remotePeerInfo.HostPort); ok {
+		if ch.onPeerStatusChanged != nil {
+			ch.onPeerStatusChanged(peer)
```

Comment:
This should be the last thing we do, since the peer itself hasn't yet processed the `connectionCloseStateChange`, and so if you query the connections, it may still contain a closed connection.

We should have caught this in our unit test by having the test query the peer properties in the callback and asserting on the number of connections.